From 681bd6b0c2cc3ff5510f2342138e40b4cf15ada9 Mon Sep 17 00:00:00 2001
From: "Bryan W. Weber" <bryan.w.weber@gmail.com>
Date: Fri, 26 Feb 2021 12:59:34 -0500
Subject: Read the version using exec during build

Reading the version directly from the version file means that runtime
dependencies do not need to be installed in the build environment.
---
 setup.py | 21 ++++++++++++---------
 1 file changed, 12 insertions(+), 9 deletions(-)

diff --git a/setup.py b/setup.py
index da875c6..601b56f 100644
--- a/setup.py
+++ b/setup.py
@@ -3,13 +3,17 @@
 
 import setuptools
 import os
+from pathlib import Path
 
-from src.ambiance.__version__ import __version__
+here = Path(__file__).parent.resolve()
+# See also: https://packaging.python.org/guides/single-sourcing-package-version/
+version = {}
+exec(here.joinpath("src", "ambiance", "__version__.py").read_text(), version)
 
 # See also: https://github.com/kennethreitz/setup.py/blob/master/setup.py
 
 NAME = 'ambiance'
-VERSION = __version__
+VERSION = version["__version__"]
 AUTHOR = 'Aaron Dettmann'
 EMAIL = 'dettmann@kth.se'
 DESCRIPTION = 'A full implementation of the ICAO standard atmosphere 1993'
@@ -23,19 +27,13 @@ README = 'README.rst'
 PACKAGE_DIR = 'src/'
 LICENSE = 'Apache License 2.0'
 
-
-here = os.path.abspath(os.path.dirname(__file__))
-
-with open(os.path.join(here, README), "r") as fp:
-    long_description = fp.read()
-
 setuptools.setup(
     name=NAME,
     version=VERSION,
     author=AUTHOR,
     author_email=EMAIL,
     description=DESCRIPTION,
-    long_description=long_description,
+    long_description=here.joinpath(README).read_text(),
     url=URL,
     include_package_data=True,
     package_dir={'': PACKAGE_DIR},
@@ -57,4 +55,9 @@ setuptools.setup(
         "Topic :: Scientific/Engineering :: Physics",
         "Topic :: Scientific/Engineering :: Atmospheric Science",
     ],
+    project_urls={
+        'Documentation': 'https://ambiance.readthedocs.io/',
+        'Source': URL,
+        'Tracker': URL + 'issues',
+    },
 )
-- 
2.30.1


From 6f7bb6fb8827da8eb8a75216bbddd4b49cbe1d4c Mon Sep 17 00:00:00 2001
From: "Bryan W. Weber" <bryan.w.weber@gmail.com>
Date: Fri, 26 Feb 2021 15:33:49 -0500
Subject: Fix building from source on Windows

Pip on Windows complains that package_dir cannot end with a slash.
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 601b56f..923785b 100644
--- a/setup.py
+++ b/setup.py
@@ -24,7 +24,7 @@ REQUIRED = [
     'scipy',
 ]
 README = 'README.rst'
-PACKAGE_DIR = 'src/'
+PACKAGE_DIR = 'src'
 LICENSE = 'Apache License 2.0'
 
 setuptools.setup(
-- 
2.30.1

